src/ostorlab/cli/scan/run/run.py

```diff
@@ -150,3 +166,53 @@ def run(
         raise click.ClickException(
             "The runtime does not support the provided agent list or group definition."
         )
+
+
+def add_cli_args_to_agent_settings(
+    agents_settings: list[definitions.AgentSettings], cli_args: list[types.AgentArg]
+) -> list[definitions.AgentSettings]:
+    """
+    Adds CLI arguments to the agent settings if they are supported by the agent.
+
+    Args:
+        agents_settings : A list of agent settings.
+        cli_args : A list of CLI Agent arguments.
+
+    Returns:
+         The updated list of agent settings.
+
+    Raises:
+        click.exceptions.Exit: If agent details are not found.
+    """
+    for agent_setting in agents_settings:
+        try:
+            agent_definition = agent_fetcher.get_definition(agent_setting.key)
+        except agent_fetcher.AgentDetailsNotFound as e:
+            console.error(e)
+            raise click.exceptions.Exit(2)
+
+        for cli_argument in cli_args:
+            supported_arg = next(
+                (
+                    arg
+                    for arg in agent_definition.args
+                    if arg.get("name") == cli_argument.name
+                ),
+                None,
+            )
+            if supported_arg is None:
+                continue
+            try:
+                agent_setting.args.append(
+                    utils_definitions.Arg.from_values(
+                        name=cli_argument.name,
+                        type=supported_arg.get("type", "string"),
+                        value=cli_argument.value,
+                    )
+                )
```

Comment:
Could have been don in the for loop for the arg instead of doing, checking None and then this code.